src/h2olog/h2olog.cc

```diff
@@ -346,6 +368,10 @@ int main(int argc, char **argv)
         probes.push_back(ebpf::USDT(h2o_pid, usdt.provider, usdt.name, usdt.probe_func));
     }
 
+    if (sampling_rate >= 0) {
+        cflags.push_back("-DH2OLOG_SAMPLING_RATE=" + std::to_string(sampling_rate));
+    }
+
```

Comment:
The tricks around `build_define_cflag` and `CFLAG_D` are fine if not too complex, though I wonder why it is define as such that it cannot be used for adding flags here.

My point is that we might want to convert the nifty tools so that they'd append entries to `std::vector<std::string>`. Then, we could use that for all the `-Dxxx=yyy` flags that we build.

Reply:
Actually, we already ready to use `build_define_cflag` here, though I cannot remember why I didn't use it.

Fixed, and I've also renamed `build_define_cflag` to `build_cflag_d` to suggest `CFLAG_D` is a wrapper for `build_cflag_d`.